src/main/java/com/minecolonies/coremod/colony/crafting/CustomRecipe.java

```diff
@@ -363,12 +363,18 @@ public boolean isValidForBuilding(IBuildingWorker building)
         IGlobalResearchTree gr = IGlobalResearchTree.getInstance();
         if (researchId != null)
         {
-            requiredEffect = colony.getResearchManager().getResearchEffects().getEffect(gr.getEffectIdForResearch(researchId), AbstractResearchEffect.class);
+            for(IResearchEffect effect : gr.getEffectsForResearch(researchId))
```

Comment:
Ok, this change doesn't make any sense. You're potentially setting the required effect multiple times, with the last one winning. I'd rather see us pick the first unlockability effect, or correctly recognize all of them.